File: 1.big_O.py
# calculating time compelxity
# (n*n)+n => n^2+n => n^2
# Time complexity O(n^2)

# def get_num(n):
#     for i in range(n):
#         for j in range(n):
#             print(i, j)

#     for k in range(7):
#         print(k)


def print_num(n):
    # O(1)
    print(n[0])


def print_str(n):
    # O(n)
    for i in range(len(n)):
        print(n[i])


def print_str_num(n, m):

    # over all
    # n+m+n^2+m = n^2+m
    # This function has the time complexity of O(n^2+m)

    # O(n+m)
    for i in range(len(n)):
        print(n[i])
    for i in range(len(m)):
        print(m[i])
    # O(n^2.m)
    for i in range(len(n)):
        for j in range(len(n)):
            for i in range(len(m)):
                print(n[i], n[j], m[i])


# print_num([10])
# print_str('Pollab')
print_str_num("nola", [1, 2, 3, 'u'])

File: 1.big_O.cpp
#include <iostream>
using namespace std;

void getBigO(int n[], int m[], size_t siz)

// over all
// n + m + n ^ 2 . m = n ^ 2 . m
// This function has the time complexity of O(n ^ 2 . m)

// O(n + m)
{
  for (int i = 0; i < siz; i++)
  {
    cout << n[i] << "\t";
  }
  cout << endl;
  for (int i = 0; i < siz; i++)
  {
    cout << m[i] << "\t";
  }
  cout << endl;
  for (int i = 0; i < siz; i++)
  {
    for (int j = 0; j < siz; j++)
    {
      for (int i = 0; i < siz; i++)
      {
        cout << n[i] << "\t" << n[j] << "\t" << m[i];
        cout << endl;
      }
    }
  }
}

int main()
{

  int num1[] = {1, 2};
  int num2[] = {6, 9};

  size_t siz = sizeof(num1) / sizeof(num1[0]);

  getBigO(num1, num2, siz);
  return 0;
}
